src/ImageSharp.Web/Middleware/ImageSharpMiddleware.cs

```diff
@@ -585,13 +567,13 @@ private ValueTask StreamDisposeAsync(Stream stream)
                         }
                         catch (Exception ex)
                         {
-                            if (!retry)
+                            if (retry)
                             {
                                 // The image has failed to be returned from the cache.
                                 // This can happen if the cached image has been physically deleted but the item is still in the LRU cache.
                                 // We'll retry running the request again in it's entirety. This ensures any changes to the source are tracked also.
                                 CacheResolverLru.TryRemove(key);
-                                await this.Invoke(httpContext);
+                                await this.Invoke(httpContext, false);
```

Comment:
I'm trying to revisit/rereview this and it's proving impossible with the additional unrelated changes. 

The logic here doesn't seem to make sense. You've reversed the condition but are calling the same method (since Invoke calls `Invoke(false)`).

As I see it the fix should be to pass `true` as the second parameter.

Reply:
I also changed the initial value of `retry` to `true` (see above), as I was assuming this boolean indicates whether it should retry the request.

But I now understand this indicates whether it is the retried request (and shouldn't retry again), so the change in https://github.com/SixLabors/ImageSharp.Web/pull/264 is indeed enough to fix it 👍🏻